src/EFCore.Sqlite.Core/Query/Internal/SqliteSqlTranslatingExpressionVisitor.cs

```diff
@@ -83,6 +83,16 @@ public class SqliteSqlTranslatingExpressionVisitor : RelationalSqlTranslatingExp
 
         protected override Expression VisitUnary(UnaryExpression unaryExpression)
         {
+            if (unaryExpression.NodeType == ExpressionType.ArrayLength
+                 && unaryExpression.Operand.Type == typeof(byte[]))
+            {
+                var sqlExpression = base.Visit(unaryExpression.Operand) as SqlExpression;
```

Comment:
nit: can be combined with the conditional operator below:

```suggestion
return base.Visit(unaryExpression.Operand) is SqlExpression operandExpression
    ? SqlExpressionFactory.Function("length", new[] { sqlExpression }, typeof(int))
    : null;
```

Reply:
can't think of why I didn't combine them, good find, thanks! Committed from the suggestion so hopefully that works out okay 🤔